src/components/PostPageDetail/index.js

```diff
@@ -112,12 +112,11 @@ const PostPageDetailIdComponent = (props) => {
   };
   const initial = () => {
     const reactionCount = item?.reaction_counts;
-    if (JSON.stringify(reactionCount) !== '{}') {
+    if (JSON.stringify(reactionCount) !== '{}' && reactionCount) {
       let count = 0;
-      const {comment} = reactionCount;
       handleVote(reactionCount);
-      if (comment !== undefined) {
-        if (comment > 0) {
+      if (reactionCount?.comment !== undefined) {
+        if (reactionCount?.comment > 0) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining here is good for preventing potential crashes due to undefined values. However, the condition can be simplified by directly checking if `reactionCount?.comment > 0`, which will return false if `reactionCount?.comment` is undefined, null, 0, NaN, or any other value that is not greater than 0.
```diff
- if (reactionCount?.comment !== undefined) {
-   if (reactionCount?.comment > 0) {
+ if (reactionCount?.comment > 0) {
```
This change simplifies the code and improves readability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->